keycloak-rest-api-extensions/src/main/java/io/cloudtrust/keycloak/email/EmailSender.java

```diff
@@ -0,0 +1,41 @@
+package io.cloudtrust.keycloak.email;
+
+import io.cloudtrust.keycloak.email.model.EmailModel;
+import io.cloudtrust.keycloak.email.provider.FreeMarkerEmailAttachmentsSenderProvider;
+import org.apache.commons.lang3.StringUtils;
+import org.keycloak.email.EmailException;
+import org.keycloak.models.KeycloakContext;
+import org.keycloak.models.KeycloakSession;
+import org.keycloak.models.RealmModel;
+import org.keycloak.models.UserModel;
+import org.keycloak.services.ErrorResponse;
+
+import javax.ws.rs.WebApplicationException;
+import javax.ws.rs.core.Response;
+import java.util.Locale;
+import java.util.Map;
+
+public class EmailSender {
+    public static Response sendMail(KeycloakSession session, RealmModel realm, EmailModel emailModel, Locale locale, UserModel user, Map<String, Object> attributes) {
+        if (emailModel.getTheming() == null && emailModel.getBasicMessage() == null) {
+            return ErrorResponse.error("Either BasicMessage or Theming configuration should be configured", Response.Status.BAD_REQUEST);
+        }
+        if (StringUtils.isBlank(emailModel.getTheming().getTemplate())) {
+            return ErrorResponse.error("Template email missing", Response.Status.BAD_REQUEST);
+        }
+        if (StringUtils.isBlank(emailModel.getTheming().getSubjectKey())) {
+            return ErrorResponse.error("Subject missing", Response.Status.BAD_REQUEST);
+        }
+
+        KeycloakContext context = session.getContext();
+        try {
+            new FreeMarkerEmailAttachmentsSenderProvider(session)
+                    .setSmtpConfig(realm.getSmtpConfig())
+                    .setDefaultLocale(locale)
+                    .send(emailModel, attributes);
+            return Response.noContent().build();
+        } catch (EmailException e) {
+            throw new WebApplicationException("Can't send mail to " + user.getEmail(), e);
```

Comment:
It seems we use the email address coming from EmailModel. The meail of the user might be a different one. Do we still need user as method parameter ?

Reply:
Removed

Reply:
Removed